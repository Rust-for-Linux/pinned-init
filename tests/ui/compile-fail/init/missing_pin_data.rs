use pin_init::*;

struct Foo {
    a: usize,
}

impl Foo {
    fn new() -> impl PinInit<Self> {
        pin_init!(Self { a: 42 })
    }
}

fn main() {}
