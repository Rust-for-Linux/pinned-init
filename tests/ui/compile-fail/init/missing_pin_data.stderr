error[E0599]: no associated item named `__pin_data` found for struct `Foo` in the current scope
 --> tests/ui/compile-fail/init/missing_pin_data.rs:9:9
  |
3 | struct Foo {
  | ---------- associated item `__pin_data` not found for this struct
...
9 |         pin_init!(Self { a: 42 })
  |         ^^^^^^^^^^^^^^^^^^^^^^^^^ associated item not found in `Foo`
  |
  = help: items from traits can only be used if the trait is implemented and in scope
  = note: the following trait defines an item `__pin_data`, perhaps you need to implement it:
          candidate #1: `HasPinData`
  = note: this error originates in the macro `pin_init` (in Nightly builds, run with -Z macro-backtrace for more info)
help: there is an associated function `__init_data` with a similar name
  |
9 -         pin_init!(Self { a: 42 })
9 +         __init_data
  |
